core/zoe/zoe/state.js

```diff
@@ -0,0 +1,46 @@
+import harden from '@agoric/harden';
+
+const makeState = issuers => {
+  const quantities = new WeakMap();
+  const offers = new WeakMap();
+  const results = new WeakMap();
+  const purses = issuers.map(issuer => issuer.makeEmptyPurse());
```

Comment:
Be aware that you will have to do an additional asynchrony dance on initialization.

Reply:
yup! That is expected